zeebe/engine/src/test/java/io/camunda/zeebe/engine/util/client/AuthorizationClient.java

```diff
@@ -271,4 +275,86 @@ public Record<AuthorizationRecordValue> delete(final String username) {
       return expectation.apply(position);
     }
   }
+
+  public static class AuthorizationUpdateClient {
+    private static final Function<Long, Record<AuthorizationRecordValue>> UPDATE_SUCCESS_SUPPLIER =
+        (position) ->
+            RecordingExporter.authorizationRecords()
+                .withIntent(AuthorizationIntent.UPDATED)
+                .withSourceRecordPosition(position)
+                .getFirst();
+    private static final Function<Long, Record<AuthorizationRecordValue>>
+        UPDATE_REJECTION_SUPPLIER =
+            (position) ->
+                RecordingExporter.authorizationRecords()
+                    .onlyCommandRejections()
+                    .withIntent(AuthorizationIntent.UPDATE)
+                    .withSourceRecordPosition(position)
+                    .getFirst();
+
+    private Function<Long, Record<AuthorizationRecordValue>> expectation = UPDATE_SUCCESS_SUPPLIER;
+    private final CommandWriter writer;
+    private final AuthorizationRecord authorizationUpdateRecord;
+    private boolean expectRejection = false;
+
+    public AuthorizationUpdateClient(final CommandWriter writer, final long authorizationKey) {
+      this.writer = writer;
+      authorizationUpdateRecord = new AuthorizationRecord().setAuthorizationKey(authorizationKey);
+    }
+
+    public AuthorizationUpdateClient withOwnerKey(final Long ownerKey) {
+      authorizationUpdateRecord.setOwnerKey(ownerKey);
+      return this;
+    }
```

Comment:
🤔 @nicpuppa should we already exclude this method since we will remove the `ownerKey` property soon?

Reply:
Yes, good catch